pkg/webhook/mutating/node/allocatable_mutator.go

```diff
@@ -41,154 +41,112 @@ func NewWebhookNodeAllocatableMutator() *WebhookNodeAllocatableMutator {
-					capacityByOvercommit := native.MultiplyResourceQuantity(core.ResourceCPU, *capacity, CPUOvercommitRatio)
-					newAllocatable = &allocatableByOvercommit
-					newCapacity = &capacityByOvercommit
+func updateNodeAnnotations(annotations map[string]string, capacity, allocatable core.ResourceList) map[string]string {
+	if annotations == nil {
+		annotations = make(map[string]string)
+	}
 
-					klog.V(6).Infof(
-						"node %s %s capacity: %v, allocatable: %v, newCapacity: %v, newAllocatable: %v",
-						node.Name, core.ResourceCPU,
-						capacity.String(), newCapacity.String(),
-						allocatable.String(), newAllocatable.String())
-				}
-			}
-		}
+	annotations[consts.NodeAnnotationOriginalCapacityCPUKey] = capacity.Cpu().String()
+	annotations[consts.NodeAnnotationOriginalCapacityMemoryKey] = capacity.Memory().String()
+	annotations[consts.NodeAnnotationOriginalAllocatableCPUKey] = allocatable.Cpu().String()
+	annotations[consts.NodeAnnotationOriginalAllocatableMemoryKey] = allocatable.Memory().String()
 
-		if newAllocatable != nil && newCapacity != nil {
-			node.Status.Allocatable[core.ResourceCPU] = *newAllocatable
-			node.Status.Capacity[core.ResourceCPU] = *newCapacity
-		}
-	}
+	return annotations
+}
 
-	memoryOvercommitRatioValue, ok := node.Annotations[consts.NodeAnnotationMemoryOvercommitRatioKey]
+// processResource Update the resource information of the node according to the specified key-value pair
+func processResource(node *core.Node, rn core.ResourceName, ratioKey, capacityKey, allocatableKey string) error {
+	ratioValue, ok := node.Annotations[ratioKey]
 	if ok {
 		var newAllocatable, newCapacity *resource.Quantity
 		// get overcommit allocatable and capacity from annotation first
-		overcommitCapacity, ok := node.Annotations[consts.NodeAnnotationOvercommitCapacityMemoryKey]
+		capacity, ok := node.Annotations[capacityKey]
 		if ok {
-			quantity, err := resource.ParseQuantity(overcommitCapacity)
+			quantity, err := resource.ParseQuantity(capacity)
 			if err != nil {
 				klog.Error(err)
 			} else {
 				newCapacity = &quantity
-				klog.V(6).Infof("node %s mem capacity by annotation: %v", node.Name, newCapacity.String())
+				klog.V(6).Infof("node %s %s capacity by annotation: %v", node.Name, newCapacity.String(), newCapacity.String())
 			}
 		}
 
-		overcommitAllocatable, ok := node.Annotations[consts.NodeAnnotationOvercommitAllocatableMemoryKey]
+		allocatable, ok := node.Annotations[allocatableKey]
 		if ok {
-			quantity, err := resource.ParseQuantity(overcommitAllocatable)
+			quantity, err := resource.ParseQuantity(allocatable)
 			if err != nil {
 				klog.Error(err)
 			} else {
 				newAllocatable = &quantity
-				klog.V(6).Infof("node %s mem allocatable by annotation: %v", node.Name, newAllocatable.String())
+				klog.V(6).Infof("node %s %s allocatable by annotation: %v", node.Name, newCapacity.String(), newAllocatable.String())
```

Comment:
The second param was used to log resource type, and the same is true for line 93